<h1>DATA ANALYTICS PROJECT</h1>

<h2>PHILIPPINES DENGUE RECORD</h2>
<h3>Inclusie year: 2016 - 2020</h3>
<h4>Analyst : Mark Eugine M. Casañada</h4>

<hr>

In [2]:
import numpy as np
import pandas as pd 
import seaborn as sns 
import matplotlib.pyplot as plt 
import warnings 
warnings.filterwarnings('ignore')

<h2>Load the dataset with the correct encoding</h2>

In [3]:
df = pd.read_csv('dataset\\Global_terrorist.csv', encoding='ISO-8859-1')

In [4]:
df.head(5)

eventid  iyear  imonth  iday approxdate  extended resolution  country  \
0  197000000001   1970       7     2        NaN         0        NaN       58   
1  197000000002   1970       0     0        NaN         0        NaN      130   
2  197001000001   1970       1     0        NaN         0        NaN      160   
3  197001000002   1970       1     0        NaN         0        NaN       78   
4  197001000003   1970       1     0        NaN         0        NaN      101   

          country_txt  region  ... addnotes scite1 scite2  scite3  dbsource  \
0  Dominican Republic       2  ...      NaN    NaN    NaN     NaN      PGIS   
1              Mexico       1  ...      NaN    NaN    NaN     NaN      PGIS   
2         Philippines       5  ...      NaN    NaN    NaN     NaN      PGIS   
3              Greece       8  ...      NaN    NaN    NaN     NaN      PGIS   
4               Japan       4  ...      NaN    NaN    NaN     NaN      PGIS   

   INT_LOG  INT_IDEO INT_MISC INT_ANY  related  
0        0         0        0       0      NaN  
1        0         1        1       1      NaN  
2       -9        -9        1       1      NaN  
3       -9        -9        1       1      NaN  
4       -9        -9        1       1      NaN  

[5 rows x 135 columns]

<h2>Filtering the data.</h2>

In [5]:
filter1 = df[df['region_txt'] == "Middle East & North Africa"]
filter2 = filter1[filter1['iyear'] >= 2000]
filter3 = filter2[filter2['attacktype1_txt'] >= 'Bombing/Explosion']
filter4 = filter3[filter3['nkill'] >= 10]

<h2>Data aggregation.</h2>

In [55]:
spatial_data1 = filter4.groupby(['longitude', 'latitude']).size().reset_index(name='terror_count')
spatial_data1['normalized_weight'] = (spatial_data1['terror_count'] - spatial_data1['terror_count'].min()) /\
                                    (spatial_data1['terror_count'].max() - spatial_data1['terror_count'].min())

In [7]:
spatial_data1

longitude   latitude  terror_count  normalized_weight
0    -8.000000  31.633333             1           0.000000
1    -7.610000  33.590000             1           0.000000
2    -0.408333  35.519444             1           0.000000
3     0.138188  35.391899             1           0.000000
4     0.719620  35.961568             1           0.000000
..         ...        ...           ...                ...
734  52.583610  29.591325             1           0.000000
735  60.582068  27.529991             1           0.000000
736  60.862701  29.483576             4           0.007481
737  61.446529  29.013931             1           0.000000
738  61.753754  26.079255             1           0.000000

[739 rows x 4 columns]

<h2>Preparing heatmap and the data.</h2>

In [8]:
import folium
from folium.plugins import HeatMap

In [90]:
spatial_list1 = spatial_data1[['longitude', 'latitude', 'normalized_weight']].values.tolist()
base_map = folium.Map(location=[30, 40], zoom_start=4)
HeatMap(spatial_list1).add_to(base_map)
base_map.save('myGeoSpatial.html')
base_map

<hr>

<h1>Regional Analysis</h1>

In [89]:
region_fil1 = df[(df['region_txt'] == 'South Asia') & (df['iyear'] >= 2000) & (df['nkill'] >= 10)]

regional_spatial = region_fil1.groupby(['longitude', 'latitude']).size().reset_index(name='terror_count')
regional_spatial['normalized_weight'] = (regional_spatial['terror_count'] - regional_spatial['terror_count'].min()) /\
                                    (regional_spatial['terror_count'].max() - regional_spatial['terror_count'].min())


regional_spatial_list = regional_spatial[['longitude', 'latitude', 'normalized_weight']].values.tolist()
regional_base_map = folium.Map(location=[30, 40], zoom_start=2)
HeatMap(regional_spatial_list).add_to(regional_base_map)
regional_base_map.save('regional.html')
regional_base_map

In [67]:
region_fil1['country_txt'].value_counts()

country_txt
Afghanistan    881
Pakistan       416
India          195
Sri Lanka       75
Nepal           27
Bangladesh       7
Name: count, dtype: int64

<h3>Insight #1 : Base on the analysis the data provided that in the south asia where kills reach 10 and above Afghanistan got the highest number of incident.</h3>

<hr>

<h1>Crime Type Analysis</h1>

In [112]:
crime_fl1 = df[(df['alternative_txt'] == 'Insurgency/Guerilla Action') & (df['iyear'] >= 2000) & (df['nkill'] >= 10)]

crime_spatial = crime_fl1.groupby(['longitude', 'latitude']).size().reset_index(name='terror_count')
crime_spatial['normalized_weight'] = (crime_spatial['terror_count'] - crime_spatial['terror_count'].min()) /\
                                    (crime_spatial['terror_count'].max() - crime_spatial['terror_count'].min())


crime_spatial_list = crime_spatial[['longitude', 'latitude', 'normalized_weight']].values.tolist()
crime_base_map = folium.Map(location=[30, 40], zoom_start=4)
HeatMap(crime_spatial_list).add_to(crime_base_map)
crime_base_map.save('crime.html')
crime_base_map

<h3>Insight #2 : Base on the analysis the data provided that in the crime type of "Insurgency/Guerilla Action" where kills reach 10 and above is mainly concentrated in the Middle East & North Africa region and Iraq got the highest number of incident record.</h3>

<hr>

<h1>Casualty Analysis</h1>

In [154]:
casualty_fl1 = df[
    (df['iyear'] >= 2000) & 
    (df['nkill'] >= 20) & 
    (df['region_txt'].isin(['Middle East & North Africa', 'Sub-Saharan Africa', 'South Asia']))]

casualty_spatial = casualty_fl1.groupby(['longitude', 'latitude']).size().reset_index(name='terror_count')
casualty_spatial['normalized_weight'] = (casualty_spatial['terror_count'] - casualty_spatial['terror_count'].min()) /\
                                    (casualty_spatial['terror_count'].max() - casualty_spatial['terror_count'].min())

casualty_spatial_list = casualty_spatial[['longitude', 'latitude', 'normalized_weight']].values.tolist()
casualty_base_map = folium.Map(location=[30, 40], zoom_start=2)
HeatMap(casualty_spatial_list).add_to(casualty_base_map)
casualty_base_map.save('casualty.html')
casualty_base_map

In [148]:
casualty_fl1['region_txt'].value_counts()

region_txt
Middle East & North Africa     1006
Sub-Saharan Africa              607
South Asia                      530
Eastern Europe                   38
Southeast Asia                   18
South America                    16
Western Europe                    9
East Asia                         9
North America                     7
Central Asia                      1
Central America & Caribbean       1
Name: count, dtype: int64

<h3>Insight #3 : Base on the analysis the data provided that in the where kills reach 20 kills casualties and above is mainly concentrated in the Middle East & North Africa region 1066 cases followed by and Sub-Saharan Africa 667 cases, and South Asia 530 cases.</h3>

<hr>

<h1>Demographic Analysis</h1>

In [155]:
demo_fl1 = df[(df['gname'] == 'Taliban') & (df['weapsubtype1_txt'] == 'Landmine')]

demo_spatial = demo_fl1.groupby(['longitude', 'latitude']).size().reset_index(name='terror_count')
demo_spatial['normalized_weight'] = (demo_spatial['terror_count'] - demo_spatial['terror_count'].min()) /\
                                    (demo_spatial['terror_count'].max() - demo_spatial['terror_count'].min())

demo_spatial_list = demo_spatial[['longitude', 'latitude', 'normalized_weight']].values.tolist()
demo_base_map = folium.Map(location=[10, 30], zoom_start=2)
HeatMap(demo_spatial_list).add_to(demo_base_map)
demo_base_map.save('demographic.html')
demo_base_map

<h3>Insight #4 : Base on the analysis the data provided that Taliban is involve there is concentrated in South Asia mainly use Landmine as weapon type.</h3>

<hr>

<h1>Target Analysis</h1>

In [167]:
target_fl1 = df[(df['targtype2_txt'] == 'Private Citizens & Property') & (df['region_txt'] == 'Middle East & North Africa')]

target_spatial = target_fl1.groupby(['longitude', 'latitude']).size().reset_index(name='terror_count')
target_spatial['normalized_weight'] = (target_spatial['terror_count'] - target_spatial['terror_count'].min()) /\
                                    (target_spatial['terror_count'].max() - target_spatial['terror_count'].min())

target_spatial_list = target_spatial[['longitude', 'latitude', 'normalized_weight']].values.tolist()
target_base_map = folium.Map(location=[10, 30], zoom_start=2)
HeatMap(target_spatial_list).add_to(target_base_map)
target_base_map.save('demographic.html')
target_base_map

<h3>Insight #5 : The analyis reveals that Private Citizens & Property attacktype mostly happen in Middle East & North Africa region.</h3>

<hr>